### ISSUE TITLE
System.Tuple aliasing works only half-way

### ISSUE THREAD

**Issue by fsoikin** [Microsoft/visualfsharp#3016]
The `System.Tuple.Create` methods are specially processed, so that their return types are considered "true" F# tuples, but the types `System.Tuple<...>` are not similarly aliased to "true" F# tuples. This leads to the following peculiar situation:

![image](https://cloud.githubusercontent.com/assets/4219105/25870114/f0304da8-34d0-11e7-83f7-ca87165226a0.png)

#### Repro steps
```
let x: System.Tuple<int,int> = System.Tuple.Create(1,2)
```

#### Expected behavior
The code should compile without error.

#### Actual behavior
Error: `FS0001	This expression was expected to have type  'System.Tuple<int,int>' but here has type ''a * 'b'`

#### Known workarounds
1. Omit type declaration:  `let x = System.Tuple.Create(1,2)`.
2. Declare type as `int * int` instead of `Tuple<int,int>`.
3. Create the tuple by calling the constructor: `let x: Tuple<int, int> = Tuple<_, _>( 1, 2 )`

**Comment by vasily-kirichenko** [comment]
I'm out of curiosity why anybody could use such code.

**Comment by fsoikin** [comment]
I found [a question about it on SO](http://stackoverflow.com/questions/43877620). This means that, at the very least, this interferes with learning F#, and therefore, with adoption of F#.

Other than that, I can imagine a real-world scenario where one is handed a structure or function expecting `System.Tuple` and is forced to use constructor (i.e. `new System.Tuple<_,_>( 1, 2 )`) to make the types agree. Calling constructor directly is the only way I could find to produce the right type. Apart from the obvious inconvenience of this solution, it is also not guaranteed to be discovered at all.

**Comment by vasily-kirichenko** [comment]
Interesting. This is also compiled:

```fsharp
let x: Tuple<int, int> = Tuple<_,_>(1, 2)
let y = 3, 4
let z = x.GetType() = y.GetType()
```

`z = true` and ILSpy shows that the types are indeed the same:
```csharp
Tuple<int, int> x = new Tuple<int, int>(1, 2);
Tuple<int, int> y = new Tuple<int, int>(3, 4);
bool z = LanguagePrimitives.HashCompare.GenericEqualityIntrinsic<Type>(x.GetType(), y.GetType());
```

**Comment by dsyme** [comment]
Yeah, we consider this by design. Direct use of the System.Tuples types does what it does, The ``(int * int)`` and ``Tuple<int,int>`` types aren't considered equivalent but the latter are translated to the former when encountered in C# metadata

**Comment by fsoikin** [comment]
@dsyme this creates an inconsistency. If the two were completely separate, that would be one thing. But with the current state of affairs, there is automatic translation one way, but not the other. This creates confusion and feeds the idea that "F# is just too confusing for newcomers".

**Comment by dsyme** [comment]
@fsoikin Perhaps.  I've never seen anyone use or teach System.Tuple directly before though.  The SO question was answered correctly with simpler code that doesn't refer to System.Tuple at all.

**Comment by dsyme** [comment]
In any case this is certainly "by design" according to the F# language spec

**Comment by nkw** [comment]
The SO question shows that the users from C# may write the following code (Or they simply copy their code from C# to F#. 
`let s = new System.Collections.Generic.Stack<Tuple<int, int>>()`
Maybe F# compiler should show some warning message that `Tuple.Create(...)` generates F# Tuple instead of `System.Tuple`?

**Comment by dsyme** [comment]
@nkw I see, yes, it's possible they will write that code.

**Comment by cartermp** [comment]
Fixed by #3283
